### DISCUSSION TITLE
Global flags --debug, --cache-dir not working

### DISCUSSION THREAD

**Discussion by aorakzai** [aquasecurity/trivy#7319] [Bugs]
### Description

Global flags --cache-dir, --debug not working on some platform and in docker

### Desired Behavior

```
trivy --cache-dir "/tmp/trivy" aws
```

Should not fail

### Actual Behavior

But it rather gives the following output

```
2024-08-08T12:26:09+02:00	FATAL	unknown flag: --cache-dir	<nil>
```

If I try to run with --debug flag then it complain about --debug flag
```
2024-08-08T12:32:27+02:00	FATAL	unknown flag: --debug	<nil>
```

### Reproduction Steps

```bash
1. install using brew install trivy
2. install aws plugin trivy plugin install github.com/aquasecurity/trivy-aws
3. run trivy --cache-dir "/tmp/trivy" aws
...
```

### Target

None

### Scanner

None

### Output Format

None

### Mode

None

### Debug Output

```bash
2024-08-08T12:32:27+02:00	FATAL	unknown flag: --debug	<nil>
```

### Operating System

macOS Sonoma

### Version

```bash
Version: 0.54.1
Vulnerability DB:
  Version: 2
  UpdatedAt: 2023-09-26 12:19:21.015897072 +0000 UTC
  NextUpdate: 2023-09-26 18:19:21.015896472 +0000 UTC
  DownloadedAt: 2023-09-26 18:06:08.173976 +0000 UTC
Check Bundle:
  Digest: sha256:24b38cdf646f0e5becf55a709ae9a3c4e819a348c28990cec0b6aabe4637d8b1
  DownloadedAt: 2023-10-06 11:07:09.944315 +0000 UTC
```

### Checklist

- [X] Run `trivy clean --all`
- [X] Read [the troubleshooting](https://aquasecurity.github.io/trivy/latest/docs/references/troubleshooting/)

**Comment by knqyf263** [comment]
@simar7 It looks like trivy-aws doesn't allow users to create an issue. Is it intentional?
https://github.com/aquasecurity/trivy-aws

**Reply by simar7** [reply]
I can't recall if I disabled them or not but I'm fine with tracking issues in either repo trivy or trivy-aws. You can go ahead and change it if so and transfer this issue over.

**Reply by knqyf263** [reply]
IMO, the plugin issues should be managed in its repository. If there is any inconvenience, we can revert it and track issues in the trivy repository.

**Reply by knqyf263** [reply]
I cannot configure trivy-aws as I don't have permission 😄

**Reply by nikpivkin** [reply]
@simar7 I don't have repo permissions either

**Reply by itaysk** [reply]
I've enabled issues on trivy-aws

**Reply by knqyf263** [reply]
<img width="479" alt="image" src="https://github.com/user-attachments/assets/6c23d54b-e9d8-4d7e-9263-1212d2fd118f">

I cannot transfer. Is this because trivy-aws doesn't enable discussions? We may need to convert it into an issue and then transfer it.

**Reply by nikpivkin** [reply]
@itaysk I converted the discussion to an issue, but still can't move it. Are any permissions required?

**Reply by simar7** [reply]
I've transferred the issue over now https://github.com/aquasecurity/trivy-aws/issues/206

**Reply by itaysk** [reply]
I've enabled discussions now

**Reply by knqyf263** [reply]
Hmm. I still cannot transfer a discussion. I guess that's because I don't have permission on trivy-aws. What if adding @aquasecurity/trivy to a collaborator?

**Reply by simar7** [reply]
Ah that would be it - sorry I've added the team now as a collaborator @knqyf263

**Reply by knqyf263** [reply]
It works now!
<img width="482" alt="image" src="https://github.com/user-attachments/assets/9f9d6a0b-5a1c-40c0-bbe0-af5ae8f23ca9">
